src/workers/hermes_worker.js

```diff
@@ -10,11 +10,18 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 import PeriodicWorker from './periodic_worker';
 
 export default class HermesWorker extends PeriodicWorker {
-  constructor(logger) {
+  constructor(dataModelEngine, config, logger) {
     super(5000, logger);
+    this.dataModelEngine = dataModelEngine;
+    this.config = config;
+    this.nonce = 0;
   }
 
   async periodicWork() {
-    this.logger.info('Hermes tick');
+    this.logger.info('Trying to upload bundle...');
+    const result = await this.dataModelEngine.finaliseBundle(this.nonce++, this.config.bundleSizeLimit, this.config.storagePeriods);
+    if (result !== null) {
+      this.logger.info('Bundle successfully uploaded');
+    }
```

Comment:
If result==null it means something is wrong and we'd like to have it logged. Additionally it'd be nice to log something like log.info({message: 'Bundle successfully uploaded', bundleId: result.bundleId})